src/modules/dialog/dlg_handlers.c

```diff
@@ -269,6 +271,25 @@ int populate_leg_info( struct dlg_cell *dlg, struct sip_msg *msg,
 
 	if (rr_set.s) pkg_free(rr_set.s);
```

Comment:
This free is in current git master at the end of this function and seems redundant now. Maybe it make sense to combine it with the other free in line 290 and move it to the end of the function?

Reply:
print_rr_body() allocates the memory for rrset.s, it is up to the caller to free it. As it may be allocated twice, it must be freed twice and that must be done before the next call to print_rr_body(). Each free is in the same scope as the call to print_rr_body(). I fail to see how these can be combined into one free.

Reply:
Thanks for the clarification